### PR TITLE
Add documentation section and update syna

### PR DESCRIPTION
**Pull request by planger**

### CODE REVIEW

File: content/documentation/_index/content.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+CDT.cloud hosts reusable components and best practices for building web-based, custom C/C++ tools based on a modern technology stack.
+Thus, CDT.cloud is not one single framework, but rather an ecosystem of platforms, components and frameworks that can be combined into a custom tool for a specific environment, e.g. an IDE for embedded software development with a specific family of boards.
+
+To learn more about building a custom tool with CDT.cloud, involved platforms, components and frameworks, please use the table of contents on the left to navigate through the documentation pages.
+
+➡️ A great starting point is the [Overview]({{< relref "overview" >}}) section.
+
+The source code of the CDT.cloud components is available on [GitHub](https://github.com/eclipse-cdt-cloud/cdt-cloud).
+With [CDT.cloud blueprint]({{< relref  "/#blueprint" >}}), there is also an example tool that demonstrates a custom C/C++ tool based on CDT.cloud.
```
Comment (sdirix) [inline content/documentation/_index/content.md:17]:
```suggestion
With [CDT.cloud Blueprint]({{< relref  "/#blueprint" >}}), there is also an example tool that demonstrates a custom C/C++ tool based on CDT.cloud.
```

File: content/documentation/overview/content.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+### How does CDT.cloud help?
+
+Custom C/C++ development tools have several reoccuring capabilities, such as creating projects, editing source code, building projects, debugging and tracing applications, etc.
+While they may differ from one specific tool use case to another, the base components can still be reused and adapted to the specifics of an environment.
+This makes building such tools significantly more efficient.
+And that's precisely where CDT.cloud is here to help: it provides a feature-rich ecosystem of reusable platforms and components that can be customized for your environment and bundled into a ready-to-use tool product for your use case.
+
+### How do I get started?
+
+In most of the cases, a great starting point is to selecting a [development tool platform]({{< relref  "tool-platform" >}}) for your custom tool, which you augment with VS Code extensions too add required tool capabilities and then put customizations for your specific environment on top.
```
Comment (sdirix) [inline content/documentation/overview/content.md:36]:
```suggestion
In most cases, a great starting point is to select a [development tool platform]({{< relref  "tool-platform" >}}) for your custom tool, which you augment with VS Code extensions too add required tool capabilities and then put customizations for your specific environment on top.
```

Comment (sdirix) [inline content/documentation/overview/content.md:36]: When I read this as a novice I don't know what is meant with "development tool platform". Clicking on the link also only refers to Theia, so I still don't know what any alternative could be. Maybe just remove this sentence completely and just start with the Blueprint?

File: content/documentation/overview/content.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+CDT.cloud also provides an example tool named [CDT.cloud blueprint]({{< relref  "blueprint" >}}), which may act as a template and a source of inspiration for your specific tool.
```
Comment (sdirix) [inline content/documentation/overview/content.md:37]:
```suggestion
CDT.cloud also provides an example tool named [CDT.cloud Blueprint]({{< relref  "blueprint" >}}), which may act as a template and a source of inspiration for your specific tool.
```

File: content/documentation/tool-platform/content.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
++++
+
+For creating a specialized C/C++ tool or custom IDE, it is typically most efficient to build on an existing tool platform, which already provides generic tool features, such as a file explorer, editors with rich language-editing support, menus, user interfaces for process output or debugging, etc.
+To meet the specific requirements of your particular tool or IDE, you then extend and customize this generic platform with specific extensions.
+
+<p align="center" style="padding-top: 1em; padding-bottom: 1em">
+  <img src="/images/cdtcloud-screenshot.png" width="85%" style="border-radius: 5px" />
+</p>
+
+As you usually want to have full control over every aspect of your custom tool, [Eclipse Theia™](https://theia-ide.org) is a popular choice for a tool platform in general, and for custom C/C++ development in particular.
```
Comment (sdirix) [inline content/documentation/tool-platform/content.md:18]: I think we should also mention VS Code and that many but not all of the CDT.cloud components can be used there.

File: content/documentation/tool-platform/content.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+Theia is an extensible and very flexible, vendor neutral open-source IDE platform to develop full-fledged multi-language Cloud & Desktop IDE-like products with state-of-the-art web technologies.
+
+### Getting started with Theia
+
+To learn more about Theia, please visit the [Theia webpage](https://theia-ide.org) and browse through its [getting started guide](https://theia-ide.org/docs/getting_started).
+This will give you a solid overview about Theia's architecture and main concepts.
+
+For your custom C/C++ tool, we recommend to start off from the [CDT.cloud Blueprint]({{< relref  "blueprint" >}}) and use it as a template.
+CDT.cloud Blueprint is a Theia-based application, which already contains a collection of extensions and configurations that likely apply to your C/C++ tool use case.
```
Comment (sdirix) [inline content/documentation/tool-platform/content.md:27]:
```suggestion
CDT.cloud Blueprint is a Theia-based application which already contains a collection of extensions and configurations that likely apply to your C/C++ tool use case.
```

File: content/documentation/tool-platform/content.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+Thus, all documentation on Theia applies to CDT.cloud Blueprint as well, but it already adds components and best practices that typically are useful for C/C++ tools.
+
+To learn more about composing Theia-based applications, in general, please refer to the [Theia documentation](https://theia-ide.org/docs/composing_applications).
```
Comment (sdirix) [inline content/documentation/tool-platform/content.md:30]:
```suggestion
To learn more about composing Theia-based applications in general, please refer to the [Theia documentation](https://theia-ide.org/docs/composing_applications).
```

File: content/documentation/tool-platform/content.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+### Extending Theia applications
+
+Theia provides typical tool features out of the box, but can be extended and customized in virtually all aspects to eventually provide a streamlined tool offering for your specific purpose.
+Therefore, Theia provides two main extension mechanisms: VS Code extensions and Theia extensions.
+
+By pulling in *VS Code extensions*, you can benefit from the vast ecosystem of tool capabilities available for VS Code.
+In the context of C/C++ tools, popular extensions include [vscode-clangd](https://open-vsx.org/extension/llvm-vs-code-extensions/vscode-clangd) for C/C++ language editing support, [GDB debug adapters](https://open-vsx.org/extension/eclipse-cdt/cdt-gdb-vscode), [CodeChecker](https://open-vsx.org/extension/codechecker/codechecker), etc.
```
Comment (sdirix) [inline content/documentation/tool-platform/content.md:38]: We don't have `CodeChecker` in Blueprint. Should we add it or remove it here?

File: content/documentation/tool-platform/content.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+*Theia extensions*, on the other hand, are more powerful than VS Code extensions and allow you to deeply customize all aspects tool platform, including the workbench structure, e.g. by adding a global toolbar, remove user interface components that you don't need in your particular tool, add wizards in the frontend or custom backend services.
```
Comment (sdirix) [inline content/documentation/tool-platform/content.md:40]:
```suggestion
*Theia extensions*, on the other hand, are more powerful than VS Code extensions and allow you to deeply customize all aspects of the tool platform, including the workbench structure, e.g. by adding a global toolbar, removing user interface components that you don't need in your particular tool, adding wizards in the frontend and providing custom backend services.
```

File: content/documentation/tool-platform/content.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+For an overview, please read more about the [Theia extension mechanisms](https://theia-ide.org/docs/extensions) and follow the guides on [integrating VS Code extensions](https://theia-ide.org/docs/authoring_vscode_extensions) and [developing Theia extensions](https://theia-ide.org/docs/authoring_extensions).
+
+### Packaging Theia applications
+
+When you want to package your custom tool as a desktop application, you'll need to build your enitre application for the operating systems you aim to support and likely want to integrate an update mechanism and add branding, such as application icons, etc.
```
Comment (sdirix) [inline content/documentation/tool-platform/content.md:46]:
```suggestion
When you want to package your custom tool as a desktop application, you'll need to build your entire application for the operating systems you aim to support and likely want to integrate an update mechanism and add branding, such as application icons, etc.
```

File: content/documentation/tool-platform/content.md [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+To learn more about dealing with those tasks, please consult the [Theia packaging documentation](https://theia-ide.org/docs/blueprint_documentation).
```
Comment (sdirix) [inline content/documentation/tool-platform/content.md:47]:
```suggestion
To learn more about dealing with those tasks, please consult the [Theia packaging documentation](https://theia-ide.org/docs/blueprint_documentation).
You can also check the CDT.cloud Blueprint for a concrete implementation.
```

File: content/documentation/overview/content.md [hunk 11]
```diff
@@ -0,0 +1,40 @@
++++
+fragment = "content"
+weight = 10
+
+title = "Overvew"
```
Comment (sdirix) [inline content/documentation/overview/content.md:5]:
```suggestion
title = "Overview"
```

File: content/documentation/blueprint/content.md [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+
+To greet and guide users on initial start-up of CDT.cloud Blueprint, it provides a welcome page.
+The welcome page is essentially a web page in a Theia view that provides general information, but also links to relevant web resources.
+Users can also trigger commands directly from the welcome page, such as opening folders, settings, as well as [creating example projects](#example-project-generator).
+
+The welcome page is contributed to the product in the [TheiaBlueprintGettingStartedContribution](https://github.com/eclipse-cdt-cloud/cdt-cloud-blueprint/blob/master/theia-extensions/theia-blueprint-product/src/browser/theia-blueprint-getting-started-contribution.ts).
+
+### Example project generator
+
+CDT.cloud Blueprint contains an example generator, which allows users to create pre-defined projects based on a project template.
```
Comment (sdirix) [inline content/documentation/blueprint/content.md:42]:
```suggestion
CDT.cloud Blueprint contains an example generator which allows users to create pre-defined projects based on a project template.
```

File: content/documentation/blueprint/content.md [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+Once a project is generated, a defined file -- such as a readme -- of this project is opened for the user.
+The generator is implemented by the Theia extension [`blueprint-example-generator`](https://github.com/eclipse-cdt-cloud/cdt-cloud-blueprint/blob/master/theia-extensions/blueprint-example-generator).
+
+The list of project types can be easily extended to add more project templates with the following steps:
+
+1. Add the project template to the folder [`resources`](https://github.com/eclipse-cdt-cloud/cdt-cloud-blueprint/tree/master/theia-extensions/blueprint-example-generator/resources)
+2. Add the project template name to the list of projects in the [command contribution](https://github.com/eclipse-cdt-cloud/cdt-cloud-blueprint/blob/master/theia-extensions/blueprint-example-generator/src/browser/example-generator-contribution.ts)
+3. Optionally specify the file to be opened after generation in the [`ExampleGeneratorServiceImpl`](https://github.com/eclipse-cdt-cloud/cdt-cloud-blueprint/blob/master/theia-extensions/blueprint-example-generator/src/node/example-generator-service.ts)
```
Comment (sdirix) [inline content/documentation/blueprint/content.md:51]: This document is very short. It seems that there is nothing in Blueprint but the welcome page and example generator. Technically this is true for the implementation but we should also mention everything which is included via a VS Code or Theia extension

File: content/documentation/additionals/content.md [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
++++
+
+Below is a list of articles on CDT.cloud:
+
+* <a target="_blank" href="https://eclipsesource.com/blogs/2021/11/26/cdt-cloud-c-c-tooling-in-the-web-cloud/">CDT.cloud? C/C++ tooling in the web/cloud</a>
+* <a target="_blank" href="https://eclipsesource.com/blogs/2022/01/10/cdt-cloud-an-overview-about-c-c-tooling-in-the-web/">CDT.cloud – An overview about C/C++ tooling in the web</a>
+
+Also, we provide a list of talk recordings and videos on CDT.cloud:
+
+* Cloud IDE Days 2022: <a target="_blank" href="https://www.youtube.com/watch?v=zRZrv4c6lhk">Building Custom C/C++ Tools in the Web</a>
```
Comment (sdirix) [inline content/documentation/additionals/content.md:18]:
```suggestion
* EclipseCon 2022: <a target="_blank" href="https://www.youtube.com/watch?v=zDDAu2LDxSs">Introducing Eclipse CDT.cloud - C/C++ tooling in the web</a>
* EclipseCon 2022: <a target="_blank" href="https://www.youtube.com/watch?v=sNdsuxE5Q1Y">What's cooking in the TraceCompass cloud project</a>
* Cloud IDE Days 2022: <a target="_blank" href="https://www.youtube.com/watch?v=zRZrv4c6lhk">Building Custom C/C++ Tools in the Web</a>
```

**Review by sdirix — COMMENTED** [review]
Already very good!

File: content/documentation/tool-platform/content.md (hunk 7 above)
Comment (planger) [reply content/documentation/tool-platform/content.md:38]: I guess we can add it at some point, but I thought I'd mention it here as an example that people may want to add. Imho not every extension mentioned here needs to be in the blueprint.

**Review by sdirix — APPROVED** [review]
LGTM

**Pull request merged**